cluster/swarm/cluster.go

```diff
@@ -241,6 +241,19 @@ func (c *Cluster) RemoveNetwork(network *cluster.Network) error {
 		for _, engine := range c.engines {
 			engine.DeleteNetwork(network)
 		}
+	} else if strings.Contains(err.Error(), "Is the docker daemon running") {
+		log.Debug("Some of the original engines unreachable - Rmoving global network from the current engines...")
+		for _, engine := range c.engines {
+			e1 := engine.RemoveNetwork(network)
+			if e1 == nil {
+				for _, engine := range c.engines {
+					engine.DeleteNetwork(network)
+				}
+				err = nil
+				break
+			}
```

Comment:
That code iterates on all of the engines until it finds an engine from which it can successfully remove the network.
If there is no Engine from which Swarm can successfully remove the network the Original error 
(in err not e1) "Error response from daemon: Cannot connect ..." will be returned to the user and nothing else will be done.
Would you expect it to behave differently in this situation?